### ISSUE TITLE
[Feature request] Add proxy setting

### ISSUE THREAD

**Issue by Nokia808** [jitsi/jitsi-meet-electron#346]
Hi. In certain situations it will be useful to use proxy to overcome blocking of Jitsi server. So, it will be very useful if you add in the "Setting" a proxy setting offer the following proxy type options:

1) none
2) HTTP,
3) SOCKS 5

The default should be off course "none" which mean direct connection without proxy.
HTTP mean HTTP type proxy
SOCKS 5 mean Socks 5 type proxy

And should be a field for address & other for port.

This case, if Jitsi is blocked by firewall, then user can set proxy at:
- SOCKS 5
- 127.0.0.1
- 9050
& this way can communicate to the server of Jitsi via Tor, overcoming firewall block ....

**Comment by saghul** [comment]
I’m not sure this is even possible in an Electron app.

If you are able to submit patches y can take a look.

**Comment by Nokia808** [comment]
@saghul 
It is possible because there is other electron applications already have proxy setting. Examples are:

1) Ferdi:
https://getferdi.com/
https://github.com/getferdi/ferdi
2) Rambox:
https://rambox.pro/
https://github.com/ramboxapp/community-edition

Unfortunately, I'm just desktop user & not programmer & learning how to programming is out of my reach, so I can not prepare a patch ....

See Ferdi because it is better than Rambox .....

**Comment by Nokia808** [comment]
@saghul 
Hi dear. I discovered the following:

https://github.com/gorhill/uBlock/wiki/Prevent-WebRTC-from-leaking-local-IP-address

As you can read: "When using a proxy, WebRTC leaks unproxied IP address even with multiple routes disabled" (reportedly fixed in Chromium 47, see comment #25)."
https://bugs.chromium.org/p/chromium/issues/detail?id=497040#c25

So, this feature request should be delayed till completely addressing the following address:
https://github.com/jitsi/jitsi-meet-electron/issues/324

Let you complete your priorities 1st, notably complete e2ee new feature & polishing it, before addressing the above 2 related issues

**Comment by Nokia808** [comment]
Data for future work:
To set over Tor network through system Tor bundle already installed & enabled on OS, user need to set in the future "proxy setting" the following:
- proxy type: SOCKS 5
- address: 127.0.0.1
- port: 9050

**Comment by Nokia808** [comment]
Hi dears
@saghul 
@Echolon 

I will edit contents in my 1st post in this issue due to discussion at https://github.com/PurpleI2P/i2pd/issues/1522
1) i2p is not suitable for chat/messenger applications (whether server dependent or serverless) unless to make them run completely hidden within i2p BUT not suitable to make them connect with clear net UNLESS by set i2p server for the chat/messenger application & this beyond scope of interest of Jitsi team. For that I will cancel request about adding support to i2p
2) the request now is just to add "proxy setting" support in this electron application. 
3) since setting Jitsi on Tor network is very simple - providing "proxy setting" will be added to Jitsi meet electron - so I will cancel request to make built-in preconfigured Tor setting because it is senseless as long as it is only:
- SOCKS 5
- 127.0.0.1
- 9050

**Comment by Nokia808** [comment]
Final note about Tor: udp not working with Tor. Tor do not support udp protocol.

**Comment by Nokia808** [comment]
I discovered this link:
https://browserleaks.com/webrtc

In the comments see what written by koxopi9069 7 months ago:
```

in cromium browsers now WebRTC Leak Local IP address hide:
Anonymize local IPs exposed by WebRTC.
Conceal local IP addresses with mDNS hostnames. – Mac, Windows
#enable-webrtc-hide-local-ips-with-mdns
```

It seem that problem of WebRTC reveal real user IP inspite of using VPN or proxy has been fixed ??
Please investigate this & do not forget to test it on Linux OS also when you have a time ....

**Comment by saghul** [comment]
PRs are welcome, but this is not something we will be working on, sorry.
